src/main/services/torrent-client.ts

```
@@ -142,6 +143,83 @@ export class TorrentClient {
             }),
           }).show();
         }
+
+        if (userPreferences?.ShutDownAfterDownloadEnabled) {
+          const dialogPromise = dialog.showMessageBox({
+            type: "question",
+            buttons: ["Cancelar", "Confirmar"],
+            defaultId: 1,
+            title: "Confirmar desligamento",
+            message: "O sistema será desligado em breve. Tem certeza de que deseja prosseguir?",
+            detail: "Se você estiver usando o computador, clique em 'Cancelar' para abortar o desligamento, caso não selecione nada dentro de alguns segudos o desligamento ocorrerá",
+            cancelId: 0,
+            noLink: true,
+            normalizeAccessKeys: false,
+          });
+          
+          const timer = setTimeout(() => {
+            userPreferencesRepository.update(
+              { id: 1 },
+              { ShutDownAfterDownloadEnabled: false }
+            ).catch(error => {
+              console.error("Erro ao redefinir o valor de ShutDownAfterDownloadEnabled:", error);
+            });
+        
+            let shutdownCommand;
+            switch (os.platform()) {
+              case 'win32':
+                shutdownCommand = 'shutdown /s /t 240'; // Comando de desligamento para Windows
+                break;
+              case 'darwin':
+                shutdownCommand = 'sudo shutdown -h +4'; // Comando de desligamento para MacOS
+                break;
+              case 'linux':
+                shutdownCommand = 'sudo shutdown -h +4'; // Comando de desligamento para Linux
+                break;
+              default:
+                console.error('Sistema operacional não suportado para desligamento.');
+                break;
+            }
+        
+            if (shutdownCommand) {
+              cp.exec(shutdownCommand, (error) => {
+                if (error) {
+                  console.error("Erro ao executar o comando de desligamento:", error);
```

No console logging, if error please just raise an exception or use `Sentry.captureException`